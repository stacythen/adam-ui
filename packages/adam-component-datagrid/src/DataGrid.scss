@import '../node_modules/ag-grid-community/src/styles/ag-grid.scss';
@import '../node_modules/ag-grid-community/src/styles/ag-theme-balham/sass/ag-theme-balham.scss';

$odd-row-background-color: #cfd8dc;
