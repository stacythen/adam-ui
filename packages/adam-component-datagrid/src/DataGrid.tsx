/* eslint-disable @typescript-eslint/explicit-function-return-type */
import React, { useCallback } from 'react';

import { AgGridReact, AgGridReactProps } from 'ag-grid-react';
import 'ag-grid-community/dist/styles/ag-grid.css';
import 'ag-grid-community/dist/styles/ag-theme-balham.css';
// import './DataGrid.scss';

import useDataGrid from './useDataGrid';
import { GridApi, GridReadyEvent, ColumnApi, PaginationChangedEvent } from 'ag-grid-community';

// // interface ISample {
// //   /** properties description here */
// //   children?: React.ReactNode;
// // }

interface ITest {
  make: string;
  model: string;
  price: string;
}

/**
 * component description here
 */
const DataGrid = (props: Partial<AgGridReactProps>): React.ReactElement => {
  const { pagination, onGridReady, onPaginationChanged, ...rest } = props;
  const [data] = useDataGrid<ITest>('https://api.myjson.com/bins/15psn9');
  let gridApi: GridApi;
  let gridColumnApi: ColumnApi;

  // // const onGridReady = useCallback(() => {
  // //   alert('onGridReady');
  // // }, [gridApi]);

  const _onGridReady = (params: GridReadyEvent) => {
    alert('onGridReady');
    gridApi = params.api;
    gridColumnApi = params.columnApi;

    // // const httpRequest = new XMLHttpRequest();
    // // const updateData = data => {
    // //   this.setState({ rowData: data });
    // // };

    // // httpRequest.open(
    // //   "GET",
    // //   "https://raw.githubusercontent.com/ag-grid/ag-grid/master/packages/ag-grid-docs/src/olympicWinnersSmall.json"
    // // );
    // // httpRequest.send();
    // // httpRequest.onreadystatechange = () => {
    // //   if (httpRequest.readyState === 4 && httpRequest.status === 200) {
    // //     updateData(JSON.parse(httpRequest.responseText));
    // //   }
    // // };
    if (onGridReady) {
      onGridReady(params);
    }
  };

  // // const _onPaginationChanged = useCallback(
  // //   () => (event: PaginationChangedEvent) => {
  // //     alert(event.api);
  // //   },
  // //   [gridApi]
  // // );

  const onPaginationChangedCallback = React.useCallback(
    event => {
      alert('_onPaginationChanged');
      alert(event.api);
      gridApi = event.api;
      if (gridApi) {
        alert(`lbLastPageFound = ${gridApi.paginationIsLastPageFound()}`);
        alert(`lbPageSize = ${gridApi.paginationGetPageSize()}`);
        alert(`lbCurrentPage = ${gridApi.paginationGetCurrentPage() + 1}`);
        alert(`lbTotalPages = ${gridApi.paginationGetTotalPages()}`);
        alert(`setLastButtonDisabled = ${gridApi.paginationIsLastPageFound()}`);
      }
      if (onPaginationChanged) {
        onPaginationChanged(event);
      }
    },
    [gridApi, onPaginationChanged]
  );

  const _onPaginationChanged = (event: PaginationChangedEvent) => onPaginationChangedCallback(event);

  const onBtFirst = useCallback(() => {
    gridApi.paginationGoToFirstPage();
  }, [gridApi]);

  const onBtLast = useCallback(() => {
    gridApi.paginationGoToLastPage();
  }, [gridApi]);

  const onBtNext = useCallback(() => {
    gridApi.paginationGoToNextPage();
  }, [gridApi]);

  const onBtPrevious = useCallback(() => {
    gridApi.paginationGoToPreviousPage();
  }, [gridApi]);

  return (
    <div
      className="ag-theme-balham"
      style={{
        height: '500px',
        width: '600px',
      }}
    >
      <AgGridReact
        {...rest}
        rowData={data}
        pagination={pagination}
        onGridReady={_onGridReady}
        onPaginationChanged={_onPaginationChanged}
      ></AgGridReact>
      {/* << < 1 2 3 > >>
        dropdown items per page
        1 - 10 of 39 items */}
      <div>
        <button onClick={onBtFirst}>{'<<'}</button>
        <button onClick={onBtPrevious}>{'<'}</button>
        123
        <button onClick={onBtNext}>{'>'}</button>
        <button onClick={onBtLast} id="btLast">
          {'>>'}
        </button>
      </div>
      <div style={{ marginTop: '6px' }}>
        <span className="label">Last Page Found:</span>
        <span className="value" id="lbLastPageFound">
          -
        </span>
        <span className="label">Page Size:</span>
        <span className="value" id="lbPageSize">
          -
        </span>
        <span className="label">Total Pages:</span>
        <span className="value" id="lbTotalPages">
          -
        </span>
        <span className="label">Current Page:</span>
        <span className="value" id="lbCurrentPage">
          -
        </span>
      </div>
    </div>
  );
};

export default DataGrid;
