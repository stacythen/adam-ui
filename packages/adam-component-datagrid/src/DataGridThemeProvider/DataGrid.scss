// Colors
$odd-row-background-color: #f3f3f3;
$border-color: #999999 !default;
$background-color: #fff !default;
$foreground-color: #777777 !default;
$header-background-color: $odd-row-background-color !default;
$header-foreground-color: $foreground-color !default;
// $icon-color: lightgray !default;
// $alt-icon-color: $background-color !default;
// $accent-color: transparent !default;
// $range-selection-background-color: $icon-color !default;

// Sizing
$grid-size: 6px !default;
// $icon-size: 12px !default;
// $header-height: $grid-size * 6 + 1 !default;
// $row-height: ($grid-size * 6 + 1) !default;
// $cell-horizontal-padding: $grid-size * 3 !default;
// $virtual-item-height: $grid-size * 5 !default;
// $header-icon-size: 14px !default;

// Misc
$row-border-width: 0px !default;
$cell-horizontal-border: 1px solid #999999aa !default;

@import 'node_modules/ag-grid-community/src/styles/ag-grid.scss';
@import 'node_modules/ag-grid-community/src/styles/ag-theme-balham/sass/ag-theme-balham.scss';

$custom-panel-item-padding: 14.5px 14px;
$custom-panel-item-margin: 0 0 0 5px;

.ag-theme-adam {
  .ag-custom-panel {
    display: flex;
    justify-content: flex-end;
    margin: 0 0 5px -5px;
  }
  .ag-custom-panel-filter-quicktext {
    padding: $custom-panel-item-padding;
    border-color: rgba(0, 0, 0, 0.23);
    border-width: 1;
  }
  .ag-custom-panel-filter-button {
    padding: $custom-panel-item-padding;
    margin: $custom-panel-item-margin;
  }
  .ag-custom-panel-sorting-button {
    padding: $custom-panel-item-padding;
    margin: $custom-panel-item-margin;
  }
}
