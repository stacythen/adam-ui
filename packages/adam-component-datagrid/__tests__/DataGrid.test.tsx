import React from 'react';
import { shallow } from 'enzyme';
import DataGrid from '../src/DataGrid';

describe('DataGrid', () => {
  it('should render properly', () => {
    const link = shallow(<DataGrid />);
    expect(link).toMatchSnapshot();
  });
});
