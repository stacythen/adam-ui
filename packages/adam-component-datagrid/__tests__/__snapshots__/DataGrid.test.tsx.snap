// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataGrid should render properly 1`] = `
<DataGridThemeProvider>
  <div
    className="adam-ui-gridwrapper"
  >
    <div
      className="ag-custom-panel"
    />
    <AgGridReact
      defaultColDef={
        Object {
          "resizable": true,
          "sortable": true,
        }
      }
      domLayout="autoHeight"
      onFilterChanged={[Function]}
      onFirstDataRendered={[Function]}
      onGridReady={[Function]}
      onGridSizeChanged={[Function]}
      onPaginationChanged={[Function]}
      onSortChanged={[Function]}
      paginationPageSize={10}
      reactNext={true}
      suppressCellSelection={true}
      suppressMovableColumns={true}
    />
  </div>
</DataGridThemeProvider>
`;
